src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +33,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToObj(n -> (n % 2 == 0) ? numbers.get(n) : numbers.get(n) - 1)
+                .filter(n -> n % 2 == 1)
+                .mapToDouble(n -> (double) n)
+                .average().getAsDouble();
```

Comment:
```suggestion
                .average()
                .getAsDouble();
```
each operation on a new line